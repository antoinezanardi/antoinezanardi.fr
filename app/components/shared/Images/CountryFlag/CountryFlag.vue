<template>
  <NuxtImg
    :alt="`${country} flag`"
    class="country-flag"
    data-bs-toggle="tooltip"
    height="25"
    loading="lazy"
    :src="`/images/flags/${countryFlagSrc}`"
    :title="$t(`Countries.${country}`)"
    width="25"
  />
</template>

<script setup lang="ts">
import type { CountryFlagProps } from "~/components/shared/Images/CountryFlag/country-flag.types";
import type { Country } from "~/models/country/country.types";

const props = defineProps<CountryFlagProps>();

const countryFlagSrc = computed<string>(() => {
  const countryFlagsSrc: Record<Country, string> = {
    france: "france-flag.webp",
    usa: "usa-flag.webp",
    canada: "canada-flag.webp",
  };

  return countryFlagsSrc[props.country];
});
</script>

<style scoped lang="scss">
.country-flag {
  width: 25px;
  height: 25px;
}
</style>