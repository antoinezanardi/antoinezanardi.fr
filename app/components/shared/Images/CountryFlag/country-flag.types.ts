import type { Country } from "~/models/country/country.types";

type CountryFlagProps = {
  country: Country;
};

export type { CountryFlagProps };