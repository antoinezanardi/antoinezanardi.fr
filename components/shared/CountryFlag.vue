<template>
  <img
    class="country-flag"
    :src="`assets/images/${countryFlagSrc}`"
    :alt="`${country} flag`"
    data-bs-toggle="tooltip"
    :title="$t(`Countries.${country}`)"
  />
</template>

<script setup lang="ts">
import { computed } from "#imports";
import type { COUNTRIES } from "~/models/Country";

const props = defineProps<{
    country: COUNTRIES;
}>();

const countryFlagSrc = computed<string>(() => {
  const countryFlagsSrc: Record<COUNTRIES, string> = {
    FRANCE: "france-flag.png",
    USA: "usa-flag.png",
    CANADA: "canada-flag.png",
  };
  return countryFlagsSrc[props.country];
});
</script>

<style scoped lang="scss">
.country-flag {
  width: 25px;
  height: 25px;
}
</style>