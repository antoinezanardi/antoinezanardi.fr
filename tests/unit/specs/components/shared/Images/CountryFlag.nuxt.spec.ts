import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import type { NuxtImg } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { CountryFlagProps } from "~/components/shared/Images/CountryFlag/country-flag.types";
import type { Country } from "~/models/country/country.types";

import CountryFlag from "@/components/shared/Images/CountryFlag/CountryFlag.vue";

describe("Country Flag Component", () => {
  let wrapper: ReturnType<typeof mount<typeof CountryFlag>>;
  const defaultProps: CountryFlagProps = {
    country: "france",
  };

  async function mountCountryFlagComponent(options: ComponentMountingOptions<typeof CountryFlag> = {}): Promise<ReturnType<typeof mount<typeof CountryFlag>>> {
    return mountSuspendedComponent(CountryFlag, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountCountryFlagComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Flag", () => {
    it("should render France flag when country is France.", () => {
      const flag = wrapper.findComponent<typeof NuxtImg>(".country-flag");

      expect(flag.attributes("src")).toBe("/images/flags/france-flag.webp");
    });

    it.each<{
      country: Country;
      expectedFlag: string;
      test: string;
    }>([
      {
        country: "france",
        expectedFlag: "/images/flags/france-flag.webp",
        test: "should render France flag when country is France.",
      },
      {
        country: "usa",
        expectedFlag: "/images/flags/usa-flag.webp",
        test: "should render USA flag when country is USA.",
      },
      {
        country: "canada",
        expectedFlag: "/images/flags/canada-flag.webp",
        test: "should render Canada flag when country is Canada.",
      },
    ])("$test", async({ country, expectedFlag }) => {
      wrapper = await mountCountryFlagComponent({ props: { country } });
      const flag = wrapper.findComponent<typeof NuxtImg>(".country-flag");

      expect(flag.attributes("src")).toBe(expectedFlag);
    });
  });
});