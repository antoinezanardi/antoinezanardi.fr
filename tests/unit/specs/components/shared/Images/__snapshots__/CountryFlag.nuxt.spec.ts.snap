// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Country Flag Component > should match snapshot when rendered. 1`] = `"<nuxt-img-stub data-v-605a4f92="" alt="france flag" format="webp" src="/images/flags/france-flag.jpg" preload="false" crossorigin="false" placeholder="false" custom="false" class="country-flag" data-bs-toggle="tooltip" title="France"></nuxt-img-stub>"`;
